### PR TITLE
Fixed #34756 -- Fixed docs HTML build on Sphinx 7.1+.

### PR DESCRIPTION
**Pull request by smithdc1**
Ticket: #[34756](https://code.djangoproject.com/ticket/34756)

See https://github.com/sphinx-doc/sphinx/commit/86b07d4a97a225e79150d14e25a768ebc4c087cc#diff-4f7d4a220c6cd7da581c080aa2eb4c0ae299ac8abcde4b8b3f8e4fe5a8ea7bdf

### CODE REVIEW

**Comment by smithdc1** [comment]
Sorry @nessita, I didn't spot that you had self assigned this when creating the ticket!

**Comment by nessita** [comment]
> Sorry @nessita, I didn't spot that you had self assigned this when creating the ticket!

No worries! I'll review tomorrow :-)

File: docs/_ext/djangodocs.py [hunk 1]
```diff
@@ -127,12 +128,26 @@ def depart_table(self, node):
...
+            self.params_left_at_level = 0
+            self.param_group_index = 0
+            # Counts as what we call a parameter group either a required
+            # parameter, or a set of contiguous optional ones.
+            self.list_is_required_param = [
+                isinstance(c, addnodes.desc_parameter) for c in node.children
+            ]
+            # How many required parameters are left.
+            self.required_params_left = sum(self.list_is_required_param)
+            self.multi_line_parameter_list = False
```
[the remark this answers is not in the document]
Comment (felixxm) [reply docs/_ext/djangodocs.py:150]: Agreed, we don't use it.

File: docs/_ext/djangodocs.py [hunk 2]
```diff
@@ -127,12 +128,26 @@ def depart_table(self, node):
...
+            # How many required parameters are left.
+            self.required_params_left = sum(
+                isinstance(c, addnodes.desc_parameter) for c in node.children
+            )
+        else:
+            self.is_first_param = True
+            self.params_left_at_level = 0
+            self.param_group_index = 0
+            # Counts as what we call a parameter group either a required
+            # parameter, or a set of contiguous optional ones.
```
Comment (felixxm) [inline docs/_ext/djangodocs.py:144]: Can you rewrite this comment to avoid "we"?

**Review by felixxm — APPROVED** [review]
@smithdc1 Thanks :+1:

**Review by nessita — COMMENTED** [review]
Looks good! Added a minor comment,

**Comment by nessita** [comment]
I noticed there is a new release of Sphinx from a few hours ago (7.1.2), but the crash is still there, so this branch is needed. Thanks!

File: docs/_ext/djangodocs.py [hunk 3]
```diff
@@ -127,12 +128,26 @@ def depart_table(self, node):
...
+            self.list_is_required_param = [
+                isinstance(c, addnodes.desc_parameter) for c in node.children
+            ]
```
[the remark this answers is not in the document]
Comment (felixxm) [reply docs/_ext/djangodocs.py:147]: Good catch :+1: Updated.

**Pull request merged**